webpack/assets/javascripts/foreman_tools.js

```diff
@@ -103,43 +103,23 @@ export function initTypeAheadSelect(input) {
 // handle table updates via turoblinks
 export function updateTable(element) {
   const uri = new URI(window.location.href);
-  let searchTerm;
-  let perPage;
-  let isSearchForm;
-  let pageNum;
 
-  if (element !== undefined) {
-    isSearchForm = element.id === 'search-form';
+  const values = { };
 
-    pageNum = $('#cur_page_num').val();
-    if (pageNum !== undefined) {
-      uri.setSearch('page', pageNum);
-    }
-
-    if (isSearchForm || element.id === 'per_page') {
-      uri.setSearch('page', 1);
-    }
-
-    if (isSearchForm) {
-      searchTerm = $(element)
-        .find('.autocomplete-input')
-        .val();
-      if (searchTerm) {
-        uri.setSearch('search', searchTerm.trim());
-      }
-    }
+  values.page = (element !== undefined) ? $('#cur_page_num').val() : undefined;
```

Comment:
We never pass undefined to the function, so i think we can get rid of all of these conditions, or just return early if the parameter is undefined if you want to be defensive.